src/main/helm/helm-repo-manager.ts

```diff
@@ -125,67 +135,87 @@ export class HelmRepoManager extends Singleton {
       const { repositories } = await this.readConfig();
 
       if (!repositories.length) {
+        if (previousAttempt) {
+          throw new Error("Previous add repo called did not add repo");
+        }
+
         await HelmRepoManager.addRepo({ name: "bitnami", url: "https://charts.bitnami.com/bitnami" });
 
-        return await this.repositories();
+        return await this.repositories(true);
       }
```

Comment:
Having recursion here seems bit odd. 

Could we just:

```typescript
const { repositories: storedRepositories } = await this.readConfig();

let repositories: HelmRepo[] = storedRepositories;

if (!repositories.length) {
  await HelmRepoManager.addRepo({
    name: "bitnami",
    url: "https://charts.bitnami.com/bitnami",
  });

  // Even better would be if we could trust that the HelmRepo used as argument of addRepo is the exact same that "readConfig" returns.
  repositories = (await this.readConfig()).repositories;
}
```

In bigger picture, I think that adding the default repository should be extracted away because now this function has two separate responsibilities.

Reply:
Yes I agree that having a default registry should be abstracted away. I have already done this in my follow up branch
 Which I will make into a PR once this is merged. 

We felt that this PR shouldn't have too many non-necessary changes.

Reply:
Good points all, but this was a change unrelated to the security fix which has effectively delayed approval of this review.